src/main/java/fi/aalto/cs/apluscourses/intellij/model/SettingsImporterImpl.java

```diff
@@ -91,11 +104,46 @@ public void importProjectSettings(@NotNull Project project, @NotNull Course cour
 
     for (String fileName : fileNames) {
       Path path = Paths.get(fileName);
+      // The ZIP contains a .idea directory with all of the settings files. We want to extract the
+      // files to the .idea directory without the .idea prefix, as otherwise we would end up with
+      // .idea/.idea/<settings_files>.
       Path pathWithoutRoot = path.subpath(1, path.getNameCount());
-      zipFile.extractFile(path.toString(), settingsPath.toString(), pathWithoutRoot.toString());
+      zipFile.extractFile(path.toString(), target.toString(), pathWithoutRoot.toString());
     }
+  }
 
-    ProjectManager.getInstance().reloadProject(project);
+  private static final String WORKSPACE_XML_COMPONENT_NAME = "CompilerWorkspaceConfiguration";
+  private static final String WORKSPACE_XML_OPTION_NAME = "AUTO_SHOW_ERRORS_IN_EDITOR";
+  private static final String WORKSPACE_XML_OPTION_VALUE = "false";
+
+  @NotNull
+  private static Document createCustomWorkspaceXml(@NotNull Path workspaceXmlPath) throws
+      IOException {
+    try {
+      Document document = DomUtil.parse(workspaceXmlPath.toFile());
+      Node projectNode = document.getDocumentElement();
+
+      // Check if a component with the given name already exists
+      Node compilerConfigurationNode = DomUtil.findChildNodeWithAttribute(projectNode,
+          "component", "name", WORKSPACE_XML_COMPONENT_NAME);
+
+      if (compilerConfigurationNode == null) {
+        Element compilerConfigurationElement = document.createElement("component");
+        compilerConfigurationElement.setAttribute("name", WORKSPACE_XML_COMPONENT_NAME);
+        compilerConfigurationNode = compilerConfigurationElement;
+        projectNode.appendChild(compilerConfigurationNode);
+      }
+
+      Element option = document.createElement("option");
```

Comment:
Is it possible that this element already exists? Are there then two of them and does that matter?

Reply:
Nice catch, there are indeed two of them. IntelliJ does some sort of `workspace.xml` "cleanup" regularly (for an example when you save the file or restart the IDE). During that process the duplicate gets removed. Another curiosity:
```<option name="AUTO_SHOW_ERRORS_IN_EDITOR" value="true" />```
gets removed entirely during the cleanup, so I assume it also removes options that have their default value.

I have just observed this cleanup process, not found it documented or anything like that, so maybe I should instead explicitly check that the option isn't there yet.

Reply:
We should refactor this hardcoded thing at some point (perhaps not until the autumn though), so if you are confident that your solution works with current version of IDEA, I'd say it's unnecessary to further improve this now.